src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/profiler/ProfilerEditingTarget.java

```diff
@@ -393,10 +486,125 @@ public void onError(ServerError error)
+               globalDisplay_.showErrorMessage("Failed to Save Profile Properties",
+                     error.getMessage());
+            }
+      });
+   }
+
+   private void saveNewFile(final String suggestedPath,
+                            final Command executeOnSuccess)
+   {
+      FileSystemItem fsi;
+      if (suggestedPath != null)
+         fsi = FileSystemItem.createFile(suggestedPath);
+      else
+         fsi = workbenchContext_.getDefaultFileDialogDir();
+ 
+      fileDialogs_.saveFile(
+            "Save File - " + getName().getValue(),
+            fileContext_,
+            fsi,
+            fileType_.getDefaultExtension(),
+            false,
+            new ProgressOperationWithInput<FileSystemItem>()
+            {
+               public void execute(final FileSystemItem saveItem,
+                                   final ProgressIndicator indicator)
+               {
+                  if (saveItem == null)
+                     return;
+
+                  workbenchContext_.setDefaultFileDialogDir(
+                        saveItem.getParentPath());
+                  
+                  final String toPath = saveItem.getPath();
+                  server_.copyProfile(
+                     getPath(),
+                     toPath,
+                     new ServerRequestCallback<JavaScriptObject>() {
+                        @Override
+                        public void onResponseReceived(JavaScriptObject response)
+                        {
+                           eventBus_.fireEvent(new FileEditEvent(saveItem));
+                           
+                           indicator.onCompleted();
+                        }
+
+                        @Override
+                        public void onError(ServerError error)
+                        {
+                           indicator.onCompleted();
+                           globalDisplay_.showErrorMessage("Failed to Save Profile",
+                                 error.getMessage());
+                        }
+                  });
+               }
+            });
+   }
 
+   private Command postSaveProfileCommand()
+   {
+      return new Command()
+      {
+         public void execute()
+         {
+         }
+      };
+   }
+   
+   private static void onMessage(String message, String file, int line)
+   {  
+      if (message == "sourcefile")
+      {
+         FilePosition filePosition = FilePosition.create(line, 0);
+         CodeNavigationTarget navigationTarget = new CodeNavigationTarget(file, filePosition);
+         
+         RStudioGinjector.INSTANCE.getFileTypeRegistry().editFile(
+               FileSystemItem.createFile(navigationTarget.getFile()),
+               filePosition);
+      }
+   }
+
+   private static native void initializeEvents() /*-{
+      var this_ = this;  
+      var handler = $entry(function(e) {
+         if (typeof e.data != 'object')
+            return;
+         if (!e.origin.startsWith(window.location.origin))
```

Comment:
Ah! GWT does replace this! It actually doesn't matter, but makes sense since this is GWT compilable.